File: Oracle/metrics/eth_price.ts
import axios from "axios";
import { Metric } from "./base";

export class ETHPriceMetric implements Metric {
    name = "eth_price";

    async fetchValue(): Promise<number | null> {
        try {
            console.log("📸 Taking snapshot of ETH price data...");
            const url =
                "https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd";
            const resp = await axios.get(url);
            const data = resp.data;

            const price = data.ethereum?.usd;
            if (price === undefined) {
                console.log(`❌ CoinGecko Error: Invalid response format ${JSON.stringify(data)}`);
                return null;
            }

            return price;
        } catch (e: any) {
            console.log(`❌ Error fetching ETH price: ${e.message}`);
            return null;
        }
    }
}

File: Oracle/python_old/metrics/eth_price.py
import requests
from .base import BaseMetric

class ETHPriceMetric(BaseMetric):
    @property
    def name(self) -> str:
        return "eth_price"

    def fetch_value(self) -> float:
        """
        Fetches current ETH price from CoinGecko.
        """
        try:
            print("📸 Taking snapshot of ETH price data...")
            url = "https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd"
            resp = requests.get(url)
            data = resp.json()
            
            price = data.get("ethereum", {}).get("usd")
            if price is None:
                print(f"❌ CoinGecko Error: Invalid response format {data}")
                return None
            
            return float(price)
        except Exception as e:
            print(f"❌ Error fetching ETH price: {e}")
            return None
